steps/flutter-version-auto-increment/0.0.1/step.yml

```diff
@@ -0,0 +1,68 @@
+title: Flutter version auto increment
+summary: |
+  Increment current version and build number in pubspec.yaml file.
+description: |
+  Increment Project current version and build number file and rewite new value in pubspec.yaml file.
+website: https://github.com/IbrahimTabba/bitrise-step-flutter-version-auto-increment
+source_code_url: https://github.com/IbrahimTabba/bitrise-step-flutter-version-auto-increment
+support_url: https://github.com/IbrahimTabba/bitrise-step-flutter-version-auto-increment/issues
+published_at: 2022-09-08T00:40:58.384109+03:00
+source:
+  git: https://github.com/IbrahimTabba/bitrise-step-flutter-version-auto-increment.git
+  commit: 28f5618cea56cb14fc63ff1f2ebb7f6a1f933829
+project_type_tags:
+- flutter
+type_tags:
+- deploy
+- build
+toolkit:
+  bash:
+    entry_file: step.sh
+inputs:
+- opts:
+    is_expand: true
+    is_required: true
+    title: Path to pubspec.yaml
+  pubspec_path: $BITRISE_SOURCE_DIR/pubspec.yaml
+- opts:
+    description: |
+      patch: X.X.0 -> X.X.1
+      minor: X.0.X -> X.1.0
+      major: 1.X.X -> 2.0.0
+    is_expand: true
+    is_required: true
+    title: Version auto increment mode
+    value_options:
+    - patch
+    - minor
+    - major
+    - none
+  version_auto_increment_mode: patch
+- increment_build_number: "yes"
```

Comment:
Suggestion: To be consistent with iOS/Android versioning, could use Bitrise build number + a configurable offset (see https://devcenter.bitrise.io/en/builds/build-numbering-and-app-versioning.html#offsetting-the-build-version).

Reply:
the source for both version and build number in flutter apps is the "version: X.X.X+X" stored in pubspec.yaml file.
this step is meant to be for production workflows, where the build number and version should be exactly one increment from the previous build, regardless of how many builds I did on bitrise and this variable BITRISE_BUILD_NUMBER

Reply:
Based on the [pubspec docs](https://dart.dev/tools/pub/pubspec#version) and [semver](https://semver.org/#spec-item-10), the build metadata (denoted by `+`) can be almost any string (git hash prefix for example) and it is not considered when comparing two versions.
My question is can we be sure that there is always a build number metadata commited? Incrementing a version without metadata (`1.2.3`) would not work. The same questoin also for a prerelease version being commited (`1.2.3-alpha`).

For reference checked an example package to understand conventions: https://pub.dev/packages/webview_flutter/versions.
Based on these, it seems that build number is usually not added, only for second build with the same minor, major and patch versions. Sometimes a prerelease version is used (`-alpha1` for example)

Reply:
An other idea: is this the input `increment_build_number` needed? It seems to me that `version_auto_increment_mode` option `none` could mean that we only increase build number. (Or add it if it was not yet added). 
The `none` option could even be renamed to `build_number_only` or similar.